engine/core/src/main/java/org/datacleaner/job/builder/AnalysisJobBuilder.java

```diff
@@ -774,9 +785,11 @@ public AnalysisJobMetadata createMetadata() {
      * 
      * @return
      * @throws IllegalStateException
-     *             if the job is invalidly configured.
+     *             if the job is invalidly configured. See
+     *             {@link #isConfigured(boolean)} for detailed exception
+     *             descriptions.
      */
-    public AnalysisJob toAnalysisJob() throws IllegalStateException {
+    public AnalysisJob toAnalysisJob() throws RuntimeException {
```

Comment:
Hmm I always learned that putting the exceptions there was nice as a way of documenting it to the next guy who reads the code. Of course there are never any guarantees that an unchecked exception cannot be thrown, but if you explicitly put it on the throws-list, then you tell the consumer of the code that it is not uncommon or that there are specific scenarios where it might fail.